thread-manager/src/lib.rs

```diff
@@ -0,0 +1,375 @@
+use {
+    anyhow::Ok,
+    serde::{Deserialize, Serialize},
+    std::{collections::HashMap, ops::Deref, sync::Arc},
+};
+
+pub mod native_thread_runtime;
+pub mod policy;
+pub mod rayon_runtime;
+pub mod tokio_runtime;
+
+pub use {
+    native_thread_runtime::{JoinHandle, NativeConfig, NativeThreadRuntime},
+    policy::CoreAllocation,
+    rayon_runtime::{RayonConfig, RayonRuntime},
+    tokio_runtime::{TokioConfig, TokioRuntime},
+};
+
+pub const MAX_THREAD_NAME_CHARS: usize = 12;
```

Comment:
why 12 instead of the pthread limit?

Reply:
12 is to keep memory layout dense and reduce max number of bytes to match (and thus to keep down the time used for matching).